src/main/kotlin/app/revanced/patcher/fingerprint/method/annotation/MethodFingerprintMetadata.kt

```diff
@@ -9,4 +9,14 @@ import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint
 @Target(AnnotationTarget.CLASS)
 annotation class FuzzyPatternScanMethod(
     val threshold: Int = 1
-)
\ No newline at end of file
+)
+
+@Target(AnnotationTarget.CLASS)
+@Retention(AnnotationRetention.SOURCE)
+annotation class ReferencedMethodDocumentation(
+    val version: String,
+    val className: String,
+    val methodName: String,
+    val parameterTypes: Array<String> = [],
+    val returnType: String
+)
```

Comment:
What is this and what use case does it have?

Reply:
> `ReferencedMethodDocumentation` is an annotation that's only present in source code and allows the developer to specify exactly which method a fingerprint targets (and in which version of the app), so it can easily be found by the developer while debugging or updating a patch

Used like this: https://github.com/1fexd/revanced-patches/blob/efea59313c47193d257065ea83deaa867171ad73/src/main/kotlin/app/revanced/patches/songpal/badge/fingerprints/ShowNotificationFingerprint.kt#L16

Reply:
And what is it used for? Why would a comment not suffice?

Reply:
It isn't used at all, I just thought it would be a good way to standardize this (and a comment just looks ugly in my opinion)

Reply:
Annotations with source retention are used for different purposes. They are used to generate code or are used by the compiler. Using such an annotation for the purpose of comments should be avoided.